packages/core/src/components/tag-input/tagInput.tsx

```diff
@@ -414,4 +415,15 @@ export class TagInput extends AbstractPureComponent<ITagInputProps, ITagInputSta
     private isValidIndex(index: number) {
         return index !== NONE && index < this.props.values.length;
     }
+
+    private handlePaste = (event: React.ClipboardEvent<HTMLInputElement>) => {
+        // we don't want it to show up in the input since we're directly adding the terms
+        event.preventDefault();
+        const { onChange, values } = this.props;
+        const clipboardText = event.clipboardData.getData("text");
+        const newValues = [].concat(...clipboardText.split(/\n|\r/).map(s => this.getValues(s)));
```

Comment:
`concat` wrapper feels wholly unnecessary. `split` and `map` each return new arrays.

Reply:
thoughts on having an `onPaste` prop, while having this be the default pasting behavior? just so consumer can plug and customize if needed?

Reply:
that is a great thought! i like this is as built-in paste behavior. but **we should be sure to invoke `inputProps.onPaste` in this method,** as this method overrides the spread on line 230 above.

Reply:
oh but yo what we should do is allow customizing the separator using the existing prop. 🤔